internal/util/debug_logging_test.go

```diff
@@ -131,22 +131,35 @@ func TestDebugLoggerThreadSafety(t *testing.T) {
 	buf := &threadSafeBuffer{buf: new(bytes.Buffer), l: &sync.RWMutex{}}
 	log := MakeDebugLoggerWithReducedRedudancy(buf, &logrus.JSONFormatter{}, 0, time.Minute*30)
 
-	// spam the logger concurrently across several goroutines to ensure no dataraces
-	wg := &sync.WaitGroup{}
-	total := 100
-	wg.Add(total)
-	for i := 0; i < total; i++ {
-		go func() {
-			defer wg.Done()
-			log.Debug("unique")
-		}()
+	const total = 100
+	writeToLogger := func() {
+		// spam the logger concurrently across several goroutines to ensure no dataraces
+		wg := &sync.WaitGroup{}
+		wg.Add(total)
+		for i := 0; i < total; i++ {
+			go func() {
+				defer wg.Done()
+				log.Debug("unique")
+			}()
+		}
+		wg.Wait()
 	}
-	wg.Wait()
-	assert.Contains(t, buf.String(), "unique")
+	assert.Eventually(t, func() bool {
```

Comment:
So the intention here is retry the test on failures? I think it is better to do the retry in a more explicit way. IMO `Eventually` is better used for waiting for something.

Reply:
Can you give an example of this more explicit way you have in mind?

Reply:
Unfortunately `assert` does not provide some utilities for retries.
I think it would like this 
```
const MaxRetries = 60
passed := false 
for i := 0; i < maxRetries; i++ {
    passed = test();
    if passed {
        break;
    }
}
assert.True(t,passed)
```

Reply:
Personally, I'd prefer not to use loops for these types of assertion in tests. Especially since we have one that IMHO is specifically designed for this: `assert|require.Eventually()`.

It my view it expresses itself clearly, that it'll do certain set of statements for such and such long and "eventually" the conditions provided have to be successful. WDYT?

Reply:
> But In my first impression, Eventually tells me that here we are waiting for a process that will take some time to make the condition satisfied, but not for some conditions that have some probability to be satisfied. What does the situation looks like here?

The situation here is that it seems that there's some probability that the log entries won't be filtered out - `len(lines) < total` condition - and hence fail the test. That's fine as long as it's sporadic. The whole point of this logger (as I understand it) is that it will filter out log entries using time window and probability.

Having said that I believe it's a pretty good fit for `.Eventually()`